### ISSUE TITLE
Find some way to display evolved CP of pokemon with multiple evolution (like Eevee)

### ISSUE THREAD

**Issue by alucas** [alucas/Snorlax#17]

**Comment by pengsloth** [comment]
I realize I am not actually helping and if there is anyway that I could (I have a bit of programming experience) I would love to. It seems to me there would be two ways to  do this. One would be if you want to see the evolved CP of Eevee it would just calculate the highest CP from its evolved forms (that would be vaporeon). Second way would be instead of writing EVONEXT or whatever into the name to tell Snorlax to evolve you could change the name to Jolteon and it would understand to give you the CP of the eevee if it evolved to jolteon. 

I have some programming experience but I don't know the interworking of this app or how it grabs information.

**Comment by andrikos** [comment]
Why not use the full range in the general case, e.g. 1650-1750, and the specific over when the evolution is certain, e.g. 1700 when the name is Sparky.

**Comment by alucas** [comment]
1. Displaying the CP of the highest evolution is not good, we need a system where we can show the CP of the Pokemon we want, event if it's a Jolteon.
2. Displaying the range is not really better, fist you need to know the different evolution, and second, if there is more than two Pokemon, the middle one will never have his CP displayed.
3. Renaming your Pokemon to the wanted evolution seem good.
4. Using easter eggs like the Sparky one is tricky (it's server side and as far as I know, can only be used once, and I have no way to know if it have already been used), but it very similar to the 3. solution.

**Comment by andrikos** [comment]
1. We could have some of the following options:
range: 1500-1600, e.g. %CPLR%
all: 1500,1550,1600, %CPLA%
max: 1600, e.g. %CPLX%
min: 1500, e.g. %CPLN%
Then the users decide what they want.
All of them are equivalent to current %CPL% when there is only one last evolution possible.
Current %CPL% could be the default of one of them (personally I would prefer the first), or keep it displaying "?" as it is now

2. We need to know the evolutions on module side (we already have this). The user needs to do some research before evolving, I think that's acceptable. For the middle one, "all:" should be ok

3-4. It can be used more than once. I have used it already 4-5 times for different evolutions and worked every single time. If you want I can try it again, I have huge number of Eevee candies.

**Comment by alucas** [comment]
I don't want to add too many %% symbol, there is already a lot and it's becoming confusing.

No need to test the Eevee trick, if naming your Pokemon Jolteon or Sparky yield the same result, there is no need to be complicated. Just use the evolution name.

**Comment by andrikos** [comment]
I don't think it will be ok with the name of the evolution (e.g. Jolteon), we need specific names up front:
- Rename as Rainer to evolve into Vaporeon
- Rename as Sparky to evolve into Jolteon
- Rename as Pyro to evolve into Flareon

What I propose for now is to have the expected CP when the evolution in known and pick up a default behavior when the evolution is unknown. Just make sure that there is something to differentiate between the two cases (e.g. a range or all values). Also, I would make the highest number come first, e.g. 1600,1550,1500 or 1600-1500, because I could expect that some people would sort with that

**Comment by alexcheveau** [comment]
I think is best to wait until the gen 2 is released. There is new multi-evo like slowbro/slowking. If they let us choose the first evo like they did with eeve we can think of implementing this on the mod like u proposed. And when the trainer don't rename i guess lucas is right, just show the ? or the code will be too much complicated

**Comment by alucas** [comment]
@andrikos I don't see the difference between renaming my eevee "Jolteon" or "Sparky". Both design the same Pokemon. Did I missed something ?

**Comment by andrikos** [comment]
If I rename an Eevee to Jolteon and then evolve, I will not get a Jolteon for sure.
I just tried and I got a Flareon.
But, if I rename an Eevee to Sparky and then evolve, I will get a Jolteon no matter what.

So, I'm proposing to show %CPL% only when the evolution is sure, i.e. single evolution pokemons, or Eevee renamed to Sparky, Pyro or Rainer. This way, the user will surely get what reported.
For generation 2 pokemon, we can do the same after we'll know the corresponding names (or whatever other mechanism they introduce)

**Comment by alexcheveau** [comment]
_"I rename an Eevee to Sparky and then evolve, I will get a Jolteon no matter what."_

You are wrong my friend. This works only for the first tume

**Comment by alucas** [comment]
@andrikos Ok I understand what you mean by "the evolution is sure". But what about all the other pokemons with two+ evolution, and no easteregg ? What is important here is for the app to be "sure" of what pokemon the trainer want to know the CP, and not what pokemon it will evolve too.

If we use that method, we can tweek it to respond to the easter egg names too.
